src/services/deep-research/DeepResearchService.ts

```diff
@@ -0,0 +1,930 @@
+	 */
+
+	public async input(content: string) {
+		if (this.isAborted()) {
+			this.publishMessage({ type: "research.error", text: "Deep research task has ended." })
+			return
+		}
+
+		this.output.push({ role: "user", content })
+
+		const stateHandlers = {
+			idle: () => this.handleIdle(content),
+			followUp: () => this.handleFollowUp(content),
+			research: () => console.log("NOOP", content),
+			done: () => this.handleDone(content),
+			aborted: () => console.log("NOOP", content),
+		} as const
+
+		console.log(`[DeepResearchService#append] executing ${this.status} handler with content = "${content}"`)
+		await stateHandlers[this.status]()
+	}
+
+	private async handleIdle(query: string) {
+		this.status = "followUp"
+		this.inquiry = { initialQuery: query, followUps: [], responses: [] }
+		this.inquiry.followUps = await this.generateFollowUps({ query })
+		await this.handleFollowUp(null)
+	}
+
+	private async handleFollowUp(content: string | null) {
+		if (content) {
+			this.inquiry.responses.push(content)
+		}
+
+		this.inquiry.responses.length >= this.inquiry.followUps.length
+			? await this.runDeepResearch()
+			: await this.publishOutput({
+					role: ResearchRole.Assistant,
+					content: this.inquiry.followUps[this.inquiry.responses.length],
+					annotations: [
+						{ type: ResearchAnnotationType.Badge, data: { label: "Follow Up", variant: "outline" } },
+					],
+				})
+	}
+
+	private async handleDone(content: string) {
+		this.messages.push({ role: "user", content })
+		const response = await this.withLoading(this.generateChatCompletion())
+
+		if (response) {
+			await this.publishOutput({ role: ResearchRole.Assistant, content: response })
+		}
+	}
+
+	/**
+	 * Statuses
+	 */
+
+	public isAborted() {
+		return this.status === "aborted"
+	}
+
+	private get status() {
+		return this._status
+	}
+
+	private set status(value: DeepResearchServiceStatus) {
+		if (this.isAborted()) {
+			return
+		}
+
+		console.log(`[setStatus] ${this.status} -> ${value}`)
+		this.publishMessage({ type: "research.status", text: JSON.stringify({ status: value }) })
+		this._status = value
+	}
+
+	/**
+	 * Actions
+	 */
+
+	public async abort() {
+		this.status = "aborted"
+		await this.saveResearchTask()
+	}
+
+	public async viewReport() {
+		const document = await this.upsertReport()
+		await vscode.window.showTextDocument(document, { preview: false })
+	}
+
+	public async createTask() {
+		const provider = this.providerRef.deref()
+
+		if (!provider) {
+			return
+		}
+
+		const document = await this.upsertReport()
+
+		if (provider) {
+			await provider.postStateToWebview()
+			await provider.postMessageToWebview({ type: "action", action: "chatButtonClicked" })
+
+			const mentionPath = vscode.workspace.workspaceFolders?.[0]
+				? `/${vscode.workspace.asRelativePath(document.uri)}`
+				: document.uri.fsPath
+
+			await provider.postMessageToWebview({
+				type: "invoke",
+				invoke: "setChatBoxMessage",
+				text: `@${mentionPath}`,
+			})
```

Comment:
Maybe not necessary, but could be nice to switch into architect mode along with this